lua/wikis/commons/Transfer/Model.lua

```diff
@@ -0,0 +1,213 @@
+		ConditionNode(ColumnName('player'), Comparator.eq, config.player),
+		ConditionTree(BooleanOperator.any):add{
+			ConditionNode(ColumnName('toteam'), Comparator.neq, ''),
+			Array.map(config.specialRoles or {}, function(role)
+				return ConditionNode(ColumnName('role2'), Comparator.eq, role)
+			end),
+		},
+	}
+
+	local records = mw.ext.LiquipediaDB.lpdb('transfer', {
+		conditions = conditions:toString(),
+		order = 'date asc',
+		limit = 5000,
+		query = 'pagename, fromteam, toteam, role1, role2, date, extradata, reference'
+	})
+
+	local transferList = Array.map(records, Transfer._processTransfer)
+
+	if ROLE_CLEAN then
+		Array.forEach(transferList, function(transfer)
+			transfer.role = ROLE_CLEAN[(transfer.role or ''):lower()]
+		end)
+	end
+
+	transferList = Array.map(transferList, FnUtil.curry(Transfer._completeTransfer, config))
+
+	-- Sort table by joinDate/leaveDate
+	table.sort(transferList, function(transfer1, transfer2)
+		if transfer1.joinDate == transfer2.joinDate then
+			if transfer1.role == LOAN and transfer2.role ~= LOAN then
+				return false
+			elseif transfer2.role == LOAN and transfer1.role ~= LOAN then
+				return true
+			end
+
+			return (transfer1.leaveDate or '') < (transfer2.leaveDate or '')
+		end
+
+		return transfer1.joinDate < transfer2.joinDate
+	end)
+
+	return transferList
+end
+
+---@param transfer transfer
+---@return TransferSpan?
+function Transfer._processTransfer(transfer)
+	local extraData = transfer.extradata
+	local transferDate = DateExt.toYmdInUtc(transfer.date)
+
+	if Logic.isNotEmpty(extraData.toteamsec) then
+		-- transfer includes multiple teams (Tl:Transfer_row |team2_2, |role2_2)
+		if (extraData.toteamsec ~= transfer.fromteam or extraData.role2sec ~= transfer.role1) and
+			(extraData.toteamsec ~= extraData.fromteamsec or extraData.role2sec ~= extraData.role1sec) then
+			-- secondary transfer
+			return {
+				team = extraData.toteamsec,
+				role = extraData.role2sec,
+				position = extraData.icon2,
+				joinDate = transferDate,
+				joinDateDisplay = extraData.displaydate or transferDate,
+				reference = {join = transfer.reference},
+			}
+		end
+
+		if (transfer.toteam ~= transfer.fromteam or transfer.role2 ~= transfer.role1) and
+			(transfer.toteam ~= extraData.fromteamsec or transfer.role2 ~= extraData.role1sec) then
+			-- primary transfer
+			return {
+				team = transfer.toteam,
+				role = transfer.role2,
+				position = extraData.icon2,
+				joinDate = transferDate,
+				joinDateDisplay = extraData.displaydate or transferDate,
+				reference = {join = transfer.reference},
+			}
+		end
+	elseif transfer.toteam ~= extraData.fromteamsec or transfer.role2 ~= extraData.role1sec then
+		-- classic transfer
+		return {
+			team = transfer.toteam,
+			role = transfer.role2,
+			position = extraData.icon2,
+			joinDate = transferDate,
+			joinDateDisplay = extraData.displaydate or transferDate,
+			reference = {join = transfer.reference},
+		}
+	end
+end
```

Comment:
Could we try to list the supported "types" of transfers in some formal way?
Currently i'm not sure why the conditions are like this, especially regarding when toteam(sec) and when fromteam(sec) is used.